File: app/src/main/java/com/antigravity/unlockly/data/model/Rule.kt
package com.antigravity.unlockly.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "rules")
data class Rule(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val productivePackages: List<String>,
    val blockedPackages: List<String>,
    val productiveMinutesTarget: Int = 30,
    val rewardMinutes: Int = 20,
    val dailyCapMinutes: Int = 120,
    val emergencyPassword: String = "",
    val isActive: Boolean = true,
    val createdAt: Long = System.currentTimeMillis()
)

File: ios/Unstuckly/Models/Rule.swift
import Foundation
import FamilyControls

public struct Rule: Identifiable, Codable {
    public var id: UUID
    public var productiveSelection: FamilyActivitySelection
    public var blockedSelection: FamilyActivitySelection
    public var productiveMinutesTarget: Int
    public var rewardMinutes: Int
    public var dailyCapMinutes: Int
    public var emergencyPassword: String
    public var isActive: Bool
    public var createdAt: Date

    public init(
        id: UUID = UUID(),
        productiveSelection: FamilyActivitySelection = FamilyActivitySelection(),
        blockedSelection: FamilyActivitySelection = FamilyActivitySelection(),
        productiveMinutesTarget: Int = 30,
        rewardMinutes: Int = 20,
        dailyCapMinutes: Int = 120,
        emergencyPassword: String = "",
        isActive: Bool = true,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.productiveSelection = productiveSelection
        self.blockedSelection = blockedSelection
        self.productiveMinutesTarget = productiveMinutesTarget
        self.rewardMinutes = rewardMinutes
        self.dailyCapMinutes = dailyCapMinutes
        self.emergencyPassword = emergencyPassword
        self.isActive = isActive
        self.createdAt = createdAt
    }
}
